easybuild/easyblocks/s/scipion.py

```diff
@@ -0,0 +1,165 @@
+            (r'^MPI_LINKERFORPROGRAMS = .*$', r'MPI_LINKERFORPROGRAMS = %s' % os.getenv('MPICXX')),
+            (r'^MPI_LIBDIR = .*$', r'MPI_LIBDIR = /'),
+            (r'^MPI_INCLUDE = .*$', r'MPI_INCLUDE = /'),
+            (r'^MPI_BINDIR = .*$', r'MPI_BINDIR = /'),
+            # The EB motioncor2 build renames that binary to an explicit name
+            (r'^MOTIONCOR2_BIN = .*$', r'MOTIONCOR2_BIN = motioncor2'),
+        ]
+        for (k, v) in map_sw2var_name.items():
+            swroot = get_software_root(k)
+            if swroot:
+                regex_subs.extend([(r'^%s = .*$' % v, r'%s = %s' % (v, swroot))])
+        cuda_root = get_software_root('CUDA')
+        if cuda_root:
+            regex_subs.extend([(r'^(.*CUDA_LIB) = .*$', r'\1 = %s/lib64' % (cuda_root))])
+            regex_subs.extend([(r'^CUDA_BIN = .*$', r'CUDA_BIN = %s/bin' % (cuda_root))])
+            regex_subs.extend([(r'^CUDA = .*$', r'CUDA = True')])
+
+        java_root = get_software_root('Java')
+        if java_root:
+            jnipath = os.path.join(java_root, 'include')
+            jnipath = os.pathsep.join([jnipath, os.path.join(jnipath, 'linux')])
+            regex_subs.extend([(r'^JNI_CPPPATH = .*$', r'JNI_CPPPATH = %s' % jnipath)])
+
+        if get_software_root('OpenCV'):
+            opencv = 'True'
+        else:
+            opencv = 'False'
+        regex_subs.extend([(r'^OPENCV = .*$', r'OPENCV = %s' % opencv)])
+
+        if get_software_root('MATLAB'):
+            matlab = 'True'
+        else:
+            matlab = 'False'
+        regex_subs.extend([(r'^MATLAB = .*$', r'MATLAB = %s' % matlab)])
+
+        apply_regex_substitutions('%s/config/scipion.conf' % self.installdir, regex_subs)
+
+    def setup_env(self):
+        # Need to setup some environemt variables before running SCons
+        env.setvar('SCIPION_HOME', self.installdir)
+        env.setvar('SCIPION_CWD', self.builddir)
+        env.setvar('SCIPION_VERSION', self.version)
+        env.setvar('SCIPION_CONFIG', '%s/config/scipion.conf' % self.installdir)
+        env.setvar('SCIPION_LOCAL_CONFIG', '%s/config/scipion.conf' % self.installdir)
+        env.setvar('SCIPION_PROTOCOLS', '%s/config/protocols.conf' % self.installdir)
+        env.setvar('SCIPION_HOSTS', '%s/config/hosts.conf' % self.installdir)
+        # Read this from scipion.conf
+        env.setvar('SCIPION_URL_SOFTWARE', 'http://scipion.cnb.csic.es/downloads/scipion/software')
+        env.setvar('LINKERFORPROGRAMS', os.getenv('CXX'))
+        env.setvar('MPI_CC', os.getenv('MPICC'))
+        env.setvar('MPI_CXX', os.getenv('MPICXX'))
+        env.setvar('MPI_LINKERFORPROGRAMS', os.getenv('MPICXX'))
+        java_root = get_software_root('Java')
+        env.setvar('JNI_CPPPATH', '%s/include:%s/include/linux' % (java_root, java_root))
+        env.setvar('JAVAC', 'javac')
+        env.setvar('JAR', 'jar')
+
+    def build_step(self):
+        """Custom build step for Scipion."""
+        self.setup_env()
```

Comment:
Why not call this once in `configure_step`?

Then you don't need to customize `build_step` at all (the environment isn't reset between steps).